consensus/tests/checkpoint.rs

```diff
@@ -5,8 +5,14 @@
 mod tests {
     use mina_consensus::{checkpoint::*, common::*};
     use mina_crypto::{hash::*, prelude::*};
+    use mina_rs_base::numbers::Length;
     use mina_rs_base::types::*;
+    use rand::{thread_rng, Rng};
```

Comment:
In my opinion tests should be deterministic and introducing randomness can break this.

If we absolutely have to use randomness we should make sure it is seeded in the code (which `ThreadRng` is not). Something like:
```rust
use rand::{SeedableRng, rngs::SmallRng};

// snip

SmallRng::seed_from_u64(SEED);
```
where `SEED` is hard-coded somewhere.

Much better would be to use hard-coded values and eliminate this dependency altogether

Reply:
It's from [here ](https://github.com/MinaProtocol/mina/blob/af76cb7980d5e81e704120290a850ea9c6f8522e/src/lib/consensus/proof_of_stake.ml#L3960) where use [ Core.Int.gen_incl](http://212.47.237.247/packages/core/v0_9_2/Core/Int/index.html)

Reply:
Won't it be weird a test is always around mid (max_blocks + min_blocks) / 2 point?
Sure the test will pass but every time you are run the same hard code value.

Reply:
Won't it be weird to use the same hard code values every time while running tests?

Reply:
The original tests in OCaml code were using a property-based testing library to generate these values, but we cannot use that here because of wasm incompatibility.

Reply:
Fixed. Not use rand at all.